### ISSUE TITLE
Consider a Promise-style `then` & `catch`

### ISSUE THREAD

**Issue by nevir** [8bitDesigner/tp-api#11]
It might be beneficial to make a breaking change to convert `then`'s interface over to just emit the found entities, and have a separate `catch` callback, so that requests adhere to `Promise`'s interface.  This would allow easy interop with `async`/`await`

**Comment by 8bitDesigner** [comment]
I totally agree, however my company no longer uses Target Process, so I don't have any way to build or test that change. If you want, however, I can make you a contributor on this project if you want to pick that task up.
